midend/hsIndexSimplify.h

```diff
@@ -0,0 +1,46 @@
+#ifndef _MIDEND_HSINDEXSIMPLIFY_H_
+#define _MIDEND_HSINDEXSIMPLIFY_H_
+
+#include "frontends/common/resolveReferences/referenceMap.h"
+#include "frontends/p4/typeChecking/typeChecker.h"
+#include "frontends/p4/typeMap.h"
+#include "ir/ir.h"
+
+namespace P4 {
+
+/// The main class for finding of non-concrete haader stack/arrya index.
```

Comment:
I would write a proper description here. For an example program. what is the before and after? What is this pass meant to do?